### PR TITLE
feat: bridge for sip call transfer not refer

### PR DESCRIPTION
**Pull request by iamprashant**
This pull request introduces major improvements to SIP bridge transfer support, audio transcoding, and session management in the telephony infrastructure. The most significant changes include adding a new bridge transfer implementation, refactoring outbound call setup, and improving audio codec handling and conversion. These changes enable seamless SIP call transfers with proper audio bridging and resource management.

**SIP Bridge Transfer and Session Management:**

* Added a new `MakeBridgeCall` and `BridgeTransfer` implementation in `sip_infra.Server`, allowing the platform to initiate a new SIP call for bridge transfers and bridge RTP audio between inbound and outbound sessions, with proper session lifecycle handling. (`api/assistant-api/sip/infra/bridge.go`)
* Refactored outbound call setup in `MakeCall` to use a shared `prepareOutboundInvite` helper, improving reliability and maintainability. Session registration and cleanup are now more robust. (`api/assistant-api/sip/infra/server.go`) [[1]](diffhunk://#diff-8dbe45afa186e42a6a8503d884682d73071ca181a741c4945754c9c676979115L1435-L1578) [[2]](diffhunk://#diff-8dbe45afa186e42a6a8503d884682d73071ca181a741c4945754c9c676979115L1602-R1497)
* Improved SIP BYE and CANCEL handling to avoid sending duplicate BYE messages by clearing the `onDisconnect` callback before ending sessions. (`api/assistant-api/sip/infra/server.go`) [[1]](diffhunk://#diff-8dbe45afa186e42a6a8503d884682d73071ca181a741c4945754c9c676979115R1069-R1071) [[2]](diffhunk://#diff-8dbe45afa186e42a6a8503d884682d73071ca181a741c4945754c9c676979115R1103-R1105)

**Audio Codec Handling and Conversion:**

* Centralized G.711 A-law/µ-law audio conversion logic in `internal_audio`, exposing `AlawToUlaw` and `UlawToAlaw` functions for use across the codebase, and updated call sites to use these utilities. (`api/assistant-api/internal/audio/audio_utils.go`, `api/assistant-api/internal/channel/telephony/internal/sip/streamer.go`) [[1]](diffhunk://#diff-dd28de3159ae02ee52002a773acdce2c4ac952540bc65e097c099e84d810c5e3R11) [[2]](diffhunk://#diff-dd28de3159ae02ee52002a773acdce2c4ac952540bc65e097c099e84d810c5e3R56-R65) [[3]](diffhunk://#diff-ed497b81a3b67eb2829be6b657800249a89766588f1dc8a25623b5546620eff6L128-R127) [[4]](diffhunk://#diff-ed497b81a3b67eb2829be6b657800249a89766588f1dc8a25623b5546620eff6L208-R205) [[5]](diffhunk://#diff-ed497b81a3b67eb2829be6b657800249a89766588f1dc8a25623b5546620eff6L283-R301)
* Added audio transcoding support in bridge transfers to automatically convert between PCMA and PCMU codecs as needed, ensuring compatibility between endpoints. (`api/assistant-api/sip/infra/bridge.go`)

**Other Improvements:**

* Improved thread safety and resource locking in SIP streamer and session management. (`api/assistant-api/internal/channel/telephony/internal/sip/streamer.go`) [[1]](diffhunk://#diff-ed497b81a3b67eb2829be6b657800249a89766588f1dc8a25623b5546620eff6L170-R181) [[2]](diffhunk://#diff-ed497b81a3b67eb2829be6b657800249a89766588f1dc8a25623b5546620eff6L283-R301)
* Enhanced event emission with additional metadata for observability. (`api/assistant-api/internal/channel/pipeline/session.go`)## Description

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Configuration change
- [ ] Refactoring (no functional changes)
- [ ] Test update
- [ ] Security fix

## Related Issues

Fixes #

### General

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) guidelines
- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

### Testing

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have tested my changes on multiple platforms (if applicable)

### Documentation

- [ ] I have updated the documentation accordingly
- [ ] I have updated the API documentation (if applicable)

### Security

- [x] My changes do not introduce any security vulnerabilities
- [x] I have not committed any sensitive data (API keys, passwords, etc.)

## Summary by CodeRabbit

* **New Features**
  * Outbound bridge calls with RTP bridging, automatic transfer initiation/status reporting, and G.711 (PCMA/PCMU) transcoding for audio interoperability.
  * Telephony events enriched with provider/direction and new transfer-request events.

* **Bug Fixes / Reliability**
  * Prevent duplicate SIP teardowns for remote-initiated hangups and avoid tearing down sessions during pending transfers; added bridge timeouts and safety handling.

* **Tests**
  * Comprehensive bridging, transcoding, lifecycle, and edge-case tests.

* **Chores**
  * Disabled telemetry in development config; dependency updates.

### CODE REVIEW

File: api/assistant-api/sip/sip.go [hunk 1]
```diff
@@ -991,10 +991,72 @@ func (m *SIPEngine) pipelineCallStart(ctx context.Context, session *sip_infra.Se
...
+	}
+
+	bridgeCtx, bridgeCancel := context.WithTimeout(inboundSession.Context(), sip_infra.BridgeCallTimeout)
+	defer bridgeCancel()
+
+	m.mu.RLock()
+	srv := m.server
+	m.mu.RUnlock()
+
+	outboundSession, err := srv.MakeBridgeCall(bridgeCtx, cfg, target, fromURI)
```
[the remark this answers is not in the document]
Comment (iamprashant) [reply api/assistant-api/sip/sip.go:1034]: @coderabbitai review

File: api/assistant-api/sip/sip.go [hunk 2]
```diff
@@ -991,10 +991,72 @@ func (m *SIPEngine) pipelineCallStart(ctx context.Context, session *sip_infra.Se
...
+	cfg := sipConfig
+	if cfg == nil {
+		cfg = inboundSession.GetConfig()
+	}
+
+	// Determine the From URI for the outbound leg
+	fromURI := cfg.CallerID
+	if fromURI == "" {
+		fromURI = cfg.Username
+	}
```
[the remark this answers is not in the document]
Comment (iamprashant) [reply api/assistant-api/sip/sip.go:1025]: @coderabbitai review
